src/client.ts

```diff
@@ -3764,6 +3764,23 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
         }
     }
 
+    public async deleteAccountData(eventType: string): Promise<void> {
+        const supportsMsc3391DeleteAccountData =
+            this.canSupport.get(Feature.AccountDataDeletion) !== ServerSupport.Unsupported;
+        // if deletion is not supported overwrite with empty content
+        if (!supportsMsc3391DeleteAccountData) {
+            await this.setAccountData(eventType, {});
+            return;
+        }
+        const path = utils.encodeUri("/user/$userId/account_data/$type", {
+            $userId: this.credentials.userId!,
+            $type: eventType,
+        });
+        return await this.http.authedRequest(Method.Delete, path, undefined, undefined, {
```

Comment:
Not sure about the unstable process: But shouldn't we here not call the endpoint depending on the `this.canSupport.get(Feature.AccountDataDeletion)` stable/unstable value?